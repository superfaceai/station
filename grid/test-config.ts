import { NockConfig } from '@superfaceai/testing';

export const nockConfig: NockConfig = {
  testInstance: expect,
};
