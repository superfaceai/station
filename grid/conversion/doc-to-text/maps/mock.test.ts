import { SuperfaceTest } from '@superfaceai/testing';

import { buildSuperfaceTest } from '../../../test-config';

describe('conversion/doc-to-text/mock', () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = buildSuperfaceTest({
      profile: 'conversion/doc-to-text',
      provider: 'mock',
    });
  });

  describe('DocToText', () => {
    // specify test case name
    it('should perform successfully', async () => {
      const result = await superface.run({
        useCase: 'ConvertDocumentToText',
        input: {
          fileName: 'big-bang.pdf',
          content: '<BinaryData>',
        },
      });

      expect(() => result.unwrap()).not.toThrow();
      expect(result).toMatchSnapshot();
    });
  });
});
