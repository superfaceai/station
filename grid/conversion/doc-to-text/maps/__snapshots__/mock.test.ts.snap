// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversion/doc-to-text/mock DocToText should perform successfully 1`] = `
Ok {
  "value": Object {
    "text": "The Big Bang event is a physical theory that describes how the universe expanded from an initial state of high density and temperature. Various cosmological models of the Big Bang explain the evolution of the observable universe from the earliest known periods through its subsequent large-scale form.",
  },
}
`;
